rustls-platform-verifier/src/tests/verification_mock/mod.rs

```diff
@@ -336,3 +366,9 @@ fn test_with_mock_root<E: std::error::Error + PartialEq + 'static>(test_case: &T
     );
     // TODO: get into specifics of errors returned when it fails.
 }
+
+enum Roots {
```

Comment:
Do you mind documenting the rationale behind this enum? At a glance its not really clear why it needs to be platform specific or the tradeoffs between it and changing `test_only_root_ca_override` to just be a boolean that these tests set instead.

Reply:
Thanks for the feedback, I've added some comments on the variant, I hope that better justify the `enum`

> changing test_only_root_ca_override to just be a boolean that these tests set instead.

It could totally be a boolean, as personal preference, I tend to avoid boolean parameters because it's not readable at call site. Especially it environment like GitHub where you don't have inlay hint, a function with boolean parameters can soon become cryptic, for instance `foo(true, false)/foo(false, true)`.

Anyway, as there is only one boolean parameter this not as impactful, I can change it to a `bool` if you want.

> its not really clear why it needs to be platform specific

I think we will get rid of this limitation when we support `new_with_extra_roots`  on all the tested platforms.